blockservice/blockservice.go

```diff
@@ -333,52 +334,39 @@ func getBlocks(ctx context.Context, ks []cid.Cid, bs blockstore.Blockstore, fget
 			return
 		}
 
-		// batch available blocks together
-		const batchSize = 32
-		batch := make([]blocks.Block, 0, batchSize)
+		var cache [1]blocks.Block // preallocate once for all iterations
 		for {
-			var noMoreBlocks bool
-		batchLoop:
-			for len(batch) < batchSize {
-				select {
-				case b, ok := <-rblocks:
-					if !ok {
-						noMoreBlocks = true
-						break batchLoop
-					}
-
-					logger.Debugf("BlockService.BlockFetched %s", b.Cid())
-					batch = append(batch, b)
-				case <-ctx.Done():
+			var b blocks.Block
+			select {
+			case v, ok := <-rblocks:
+				if !ok {
 					return
-				default:
-					break batchLoop
 				}
+				b = v
```

Comment:
```suggestion
				cache[0] = v
```
maybe?

Reply:
But then we would need to write `bs.Put(ctx, cache[0])` which isn't pretty.